test/e2e/service_test.go

```diff
@@ -81,3 +82,60 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 
 	return svc, cleanup
 }
+
+// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
+// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+func TestNodePortWindows(t *testing.T) {
+	skipIfNoWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svcName := "agnhost"
+	if len(clusterInfo.windowsNodes) == 0 {
+		t.Fatalf("The cluster should have at least one Windows Node")
+	}
```

Comment:
this feels redundant with `skipIfNoWindowsNodes(t)`

Reply:
Removed.